hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -82,14 +84,18 @@
   static String LOCATE_PREFETCH_LIMIT = "hbase.client.locate.prefetch.limit";
 
   private static final int DEFAULT_LOCATE_PREFETCH_LIMIT = 10;
+  private static final long GET_META_LOCATIONS_TIMEOUT = 2000; // 2 seconds
```

Comment:
What's this hard coded timeout here?

Reply:
This is the timeout value to wait for meta locations from registry.

Reply:
Moving it to "hbase.rpc.read.timeout".